src/_pytest/nodes.py

```diff
@@ -134,19 +136,86 @@ def _getcustomclass(self, name):
     def __repr__(self):
         return "<%s %r>" % (self.__class__.__name__, getattr(self, "name", None))
 
-    def warn(self, code, message):
-        """ generate a warning with the given code and message for this
-        item. """
+    def warn(self, code_or_warning, message=None):
```

Comment:
for full backward compatibility we need to accept `code` as keyword argument as well (in which case `code_or_warning` should be `None`

Reply:
Done, but that means we will need to keep `code_or_warning` as a valid keyword argument indefinitely right?

Reply:
lets add an underscore - it shouldnt be public in that way

lets also document it a backward compatibility element that wil lbe removed with the legacy warning support

Reply:
Good idea, done!